p2p/transport.go

```diff
@@ -14,40 +14,103 @@ const (
 	defaultProtocol Protocol = MConnProtocol
 )
 
-// Transport is an arbitrary mechanism for exchanging bytes with a peer.
+// Protocol identifies a transport protocol.
+type Protocol string
+
+// Transport is a connection-oriented mechanism for exchanging data with a peer.
 type Transport interface {
-	// Accept waits for the next inbound connection on a listening endpoint. If
-	// this returns io.EOF or ErrTransportClosed the transport should be
-	// considered closed and further Accept() calls are futile.
+	// Stringer is used to display the transport, e.g. in logs.
+	fmt.Stringer
```

Comment:
We use this to log the transport. Without it, the logger will use reflection to output the struct fields, and that triggers the race detector and can cause panics since it accesses fields tht are written concurrently.